cpp/ql/src/semmle/code/cpp/dataflow/TaintTracking.qll

```diff
@@ -249,10 +249,18 @@ module TaintTracking {
         ) or (
           inModel.isInQualifier() and
           exprIn = call.getQualifier()
+        ) or (
+          inModel.isInReturnValue() and
+          exprIn = call
         )
       ) and (
-        outModel.isOutReturnValue() and
-        exprOut = call
+        (
+          outModel.isOutReturnValue() and
+          exprOut = call
+        ) or exists(int argOutIndex |
+          outModel.isOutParameterPointer(argOutIndex) and
+          exprOut = call.getArgument(argOutIndex)
```

Comment:
Can we avoid adding flow from an out-parameter-pointer to an (input) argument? That seems backwards, and I don't think that flow will go anywhere since we have def-use flow and not use-use. What case is it needed for?